bundle/config/root_test.go

```diff
@@ -252,3 +254,30 @@ func TestIsFullVariableOverrideDef(t *testing.T) {
 		assert.Equal(t, tc.expected, isFullVariableOverrideDef(tc.value), "test case %d", i)
 	}
 }
+
+const inputJob = `{"deployment":{"kind":"BUNDLE","metadata_file_path":"/Workspace/Users/[USERNAME]/.bundle/test-bundle/default/state/metadata.json"},"edit_mode":"UI_LOCKED","format":"MULTI_TASK","max_concurrent_runs":1,"name":"job foo","permissions":[],"queue":{"enabled":true},"tasks":[{"run_job_task":{"job_id":"${resources.jobs.bar.id}"},"task_key":"job_task"}]}`
+
+// I'm expecting either:
+// a) job_id is ignored, everything else is parsed
+// b) an error
+// However, what happens is task_key gets zeroed out:
+// const expectedOut = `{"deployment":{"kind":"BUNDLE","metadata_file_path":"/Workspace/Users/[USERNAME]/.bundle/test-bundle/default/state/metadata.json"},"edit_mode":"UI_LOCKED","format":"MULTI_TASK","max_concurrent_runs":1,"name":"job foo","permissions":[],"queue":{"enabled":true},"tasks":[{"run_job_task":{"job_id":0},"task_key":"job_task"}]}`
+const actualOut = `{"deployment":{"kind":"BUNDLE","metadata_file_path":"/Workspace/Users/[USERNAME]/.bundle/test-bundle/default/state/metadata.json"},"edit_mode":"UI_LOCKED","format":"MULTI_TASK","lifecycle":{},"max_concurrent_runs":1,"name":"job foo","queue":{"enabled":true},"tasks":[{"run_job_task":{"job_id":0},"task_key":""}]}`
+
+func TestUnmarshalResourceJob(t *testing.T) {
+	x := resources.Job{}
+	err := json.Unmarshal([]byte(inputJob), &x)
+	require.NoError(t, err)
```

Comment:
Do you know if/where this error gets swallowed?

Reply:
No, I did not debug this further than this.